src/MatrixEventHandler.ts

```diff
@@ -250,46 +251,74 @@ export class MatrixEventHandler {
         }
     }
 
+    public async onTyping(r: Request<TypingEvent>) {
+        const typing = r.getData();
+        const ctx: RoomBridgeStoreEntry = (await this.store.getRoomEntryByMatrixId(typing.room_id) || {
+            matrix: undefined,
+            remote: undefined,
+            data: {},
+        });
+        const roomType: string|null = ctx.matrix ? ctx.matrix.get("type") : null;
+
+        if (roomType === MROOM_TYPE_IM) {
```

Comment:
node-purple only supports typing in IMs, in the future we should expand this to group rooms for xmpp.js's sake.